### PR TITLE
client/tailscale: make a helper for json.Marshal'ed request bodies

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

**Review by maisem — APPROVED** [review]

File: client/tailscale/localclient.go [hunk 1]
```diff
@@ -920,3 +911,24 @@ func tailscaledConnectHint() string {
 	}
 	return "not running?"
 }
+
+type jsonReader struct {
+	b   *bytes.Reader
+	err error // sticky JSON marshal error, if any
+}
+
+// jsonBody returns an io.Reader that marshals v as JSON and then reads its.
```
Comment (shayne) [inline client/tailscale/localclient.go:920]: s/its/it/

File: client/tailscale/localclient.go [hunk 2]
```diff
@@ -920,3 +911,24 @@ func tailscaledConnectHint() string {
 	}
 	return "not running?"
 }
+
+type jsonReader struct {
+	b   *bytes.Reader
+	err error // sticky JSON marshal error, if any
```
Comment (shayne) [inline client/tailscale/localclient.go:917]: 🥇

**Review by shayne — APPROVED** [review]
This is great! Small typo nit

File: client/tailscale/localclient.go (hunk 1 above)
Comment (bradfitz) [reply client/tailscale/localclient.go:920]: I originally wrote "its JSON" and thought that was redundant.

**Pull request merged**